sympy/integrals/deltafunctions.py

```diff
@@ -55,14 +55,14 @@ def change_mul(node, x):
         if arg.is_Pow and arg.base.func is DiracDelta:
             new_args.append(arg.func(arg.base, arg.exp - 1))
             arg = arg.base
-        if dirac is None and (arg.func is DiracDelta and arg.is_simple(x)):
+        if dirac is None and (isinstance(arg, DiracDelta) and arg.is_simple(x)):
             dirac = arg
         else:
             new_args.append(arg)
     if not dirac:  # there was no simple dirac
         new_args = []
         for arg in sorted_args:
-            if arg.func is DiracDelta:
+            if isinstance(arg, DiracDelta):
                 new_args.append(arg.expand(diracdelta=True, wrt=x))
             elif arg.is_Pow and arg.base.func is DiracDelta:
```

Comment:
It is perhaps worth checking again for `.func is ` to see if any others (like this one) got missed.